tests/unit/createBirthdaySquad.test.ts

```diff
@@ -0,0 +1,254 @@
+import "@/testUtils/unit/mockDb";
+import "@/testUtils/unit/mockEventBridge";
+
+import {
+  EventBridgeClient,
+  PutEventsCommand,
+} from "@aws-sdk/client-eventbridge";
+import dayjs from "dayjs";
+import utc from "dayjs/plugin/utc";
+import type { Mock } from "vitest";
+import { afterEach, beforeEach, describe, expect, it, vi } from "vitest";
+
+import * as getSquadMembers from "@/db/queries/getSquadMembers";
+import { squadJoins, users } from "@/db/schema";
+import type { Events } from "@/events";
+import { handler } from "@/functions/events/createBirthdaySquad";
+import { BIRTHDAY_SQUAD_SIZE } from "@/functions/utils/constants";
+import { openConversation } from "@/services/slack/openConversation";
+import { seedSquadJoins } from "@/testUtils/seedSquadJoins";
+import { testDb } from "@/testUtils/testDb";
+import { mockEventBridgePayload } from "@/testUtils/unit/mockEventBridgePayload";
+import { sendMockSqsMessage } from "@/testUtils/unit/sendMockSqsMessage";
+
+dayjs.extend(utc);
+
+const constants = vi.hoisted(() => ({
+  birthdayPerson: "U001",
+  eventId: "E001",
+  teamId: "T001",
+  otherUserIds: ["U002", "U003", "U004", "U005", "U006"],
+  converastionId: "CH001",
+}));
+
+vi.mock("@/services/slack/openConversation", async () => ({
+  openConversation: vi.fn().mockResolvedValue(constants.converastionId),
+}));
+
+describe("CreateBirthdaySquad", () => {
+  let openConversationMock: Mock;
+
+  beforeEach(() => {
+    openConversationMock = openConversation as Mock;
+  });
+
+  afterEach(async () => {
+    vi.clearAllMocks();
+  });
+
+  it("Should call getSquadMembers", async () => {
+    const event = {
+      birthdayPerson: constants.birthdayPerson,
+      eventId: constants.eventId,
+      team: constants.teamId,
+    } satisfies Events["createBirthdaySquad"];
+    const getSquadMembersSpy = vi.spyOn(getSquadMembers, "getSquadMembers");
+
+    openConversationMock.mockResolvedValueOnce(constants.converastionId);
+
+    await sendMockSqsMessage("createBirthdaySquad", event, handler);
+
+    expect(getSquadMembersSpy).toHaveBeenCalledWith(
+      constants.teamId,
+      constants.birthdayPerson,
+    );
+  });
+});
+
+describe("Final squad size is less then 2", () => {
```

Comment:
I am not 100% sure on this, but imo different test suits should be for different things. Here its the same function so I think all tests should be in one describe block.